app/src/main/java/app/revanced/integrations/patches/AlternativeThumbnailsPatch.java

```diff
@@ -18,25 +25,287 @@
+            LogHelper.printDebug(() -> "Replacement url: " + sanitizedReplacementUrl);
+
+            return includeTracking
+                    ? sanitizedReplacementUrl + decodedUrl.viewTrackingParameters
+                    : sanitizedReplacementUrl;
+        } catch (Exception ex) {
+            LogHelper.printException(() -> "overrideImageURL failure", ex);
+            return originalUrl;
+        }
+    }
+
+    /**
+     * Injection point.
+     * <p>
+     * Cronet considers all completed connections as a success, even if the response is 404 or 5xx.
+     */
+    public static void handleCronetSuccess(UrlRequest request, @NonNull UrlResponseInfo responseInfo) {
+        try {
+            final int statusCode = responseInfo.getHttpStatusCode();
+            if (statusCode != 200) {
+                String url = responseInfo.getUrl();
+
+                if (usingDeArrow() && urlIsDeArrow(url)) {
+                    LogHelper.printDebug(() -> "handleCronetSuccess, statusCode: " + statusCode);
+                    handleDeArrowError(url, statusCode);
+                    return;
+                }
+
+                if (usingVideoStills() && statusCode == 404) {
+                    // Fast alt thumbnails is enabled and the thumbnail is not available.
+                    // The video is:
+                    // - live stream
+                    // - upcoming unreleased video
+                    // - very old
+                    // - very low view count
+                    // Take note of this, so if the image reloads the original thumbnail will be used.
+                    DecodedThumbnailUrl decodedUrl = DecodedThumbnailUrl.decodeImageUrl(url);
+                    if (decodedUrl == null) {
+                        return; // Not a thumbnail.
+                    }
+
+                    LogHelper.printDebug(() -> "handleCronetSuccess, image not available: " + url);
+
+                    ThumbnailQuality quality = ThumbnailQuality.altImageNameToQuality(decodedUrl.imageQuality);
+                    if (quality == null) {
+                        // Video is a short or a seekbar thumbnail, but somehow did not load.  Should not happen.
+                        LogHelper.printDebug(() -> "Failed to recognize image quality of url: " + decodedUrl.sanitizedUrl);
+                        return;
+                    }
+
+                    VerifiedQualities.setAltThumbnailDoesNotExist(decodedUrl.videoId, quality);
+                }
+            }
+        } catch (Exception ex) {
+            LogHelper.printException(() -> "Callback success error", ex);
+        }
+    }
+
+    /**
+     * Injection point.
+     *
+     * To test failure cases, try changing the API URL to each of:
+     * - A non existent domain.
+     * - A url path of something incorrect (ie: /v1/nonExistentEndPoint).
+     *
+     * Known limitation: YT uses an infinite timeout, so this hook is never called if a host never responds.
```

Comment:
Would it be possible to artificially start a timeout when a request is successfully made? This way, even if the host goes down, the timeout could handle the failure.